### PR TITLE
Add VkLayer_GF_shader_fuzzer

### PR DESCRIPTION
**Pull request by paulthomson**
Adds a layer that fuzzes shaders using spirv-fuzz. This change required many dependencies to be added and many changes to the CMake.

### CODE REVIEW

File: src/VkLayer_GF_shader_fuzzer/VkLayer_GF_frame_counter_windows.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (afd) [inline src/VkLayer_GF_shader_fuzzer/VkLayer_GF_frame_counter_windows.json]: I think perhaps this file is wrongly-named (and I guess has the wrong contents) - should it be VkLayer_GF_shader_fuzzer_windows.json?

**Review by afd — CHANGES_REQUESTED** [review]
I think one .json file is wrong.  I didn't yet review the layer code.

File: src/VkLayer_GF_shader_fuzzer/src/shader_fuzzer_layer.cc [hunk 2]
```diff
@@ -0,0 +1,784 @@
...
+GlobalData global_data_;  // NOLINT(cert-err58-cpp)
+#pragma clang diagnostic pop
+
+GlobalData* GetGlobaData() { return &global_data_; }
+
+const std::array<VkLayerProperties, 1> kLayerProperties{{{
+    "VkLayer_GF_shader_fuzzer",     // layerName
+    VK_MAKE_VERSION(1U, 1U, 130U),  // specVersion NOLINT(hicpp-signed-bitwise)
+    1,                              // implementationVersion
+    "Frame counter layer.",         // description
```
Comment (afd) [inline src/VkLayer_GF_shader_fuzzer/src/shader_fuzzer_layer.cc]: Description needs updating.

**Review by afd — COMMENTED** [review]
Looks really good; I noticed one small thing.

File: src/VkLayer_GF_shader_fuzzer/VkLayer_GF_frame_counter_windows.json (hunk 1 above)
Comment (paulthomson) [reply src/VkLayer_GF_shader_fuzzer/VkLayer_GF_frame_counter_windows.json]: Thanks! The correct file is there. I should have deleted this file!

**Review by afd — APPROVED** [review]
LGTM!

**Pull request merged**